src/game.cpp

```diff
@@ -1788,7 +1792,20 @@ void Game::run()
 		updateProfilers(runData, stats, draw_times, dtime);
 		processUserInput(&flags, &runData, dtime);
 		// Update camera before player movement to avoid camera lag of one frame
-		updateCameraDirection(&cam_view, &flags);
+		updateCameraDirection(&cam_view_target, &flags);
+		float cam_smoothing = 0;
+		if (g_settings->getBool("cinematic"))
+			cam_smoothing = 1 - g_settings->getFloat("cinematic_camera_smoothing");
+		else
+			cam_smoothing = 1 - g_settings->getFloat("camera_smoothing");
+		if (cam_smoothing <= 0)
+			cam_smoothing = 0.01;
+		if (cam_smoothing > 1)
```

Comment:
else if (not just if... both conditions can't be true in this case)

You could also use `cam_smoothing = rangelim(cam_smoothing, 0.01f, 1.0f);` (see util/numeric.h)